packages/client/components/SpotlightGroups.tsx

```diff
@@ -1,67 +1,109 @@
 })
 
 interface Props {
-  meeting: SpotlightGroups_meeting$key
   phaseRef: RefObject<HTMLDivElement>
-  viewer: SpotlightGroups_viewer$key
+  queryRef: PreloadedQuery<SpotlightGroupsQuery>
 }
 
 const SpotlightGroups = (props: Props) => {
-  const {phaseRef} = props
-  const userData = useFragment(
+  const {phaseRef, queryRef} = props
+  const data = usePreloadedQuery<SpotlightGroupsQuery>(
     graphql`
-      fragment SpotlightGroups_viewer on User {
-        similarReflectionGroups(reflectionId: $reflectionId, searchQuery: $searchQuery) {
-          id
-          ...ReflectionGroup_reflectionGroup
+      query SpotlightGroupsQuery($reflectionId: ID!, $searchQuery: String!, $meetingId: ID!) {
+        viewer {
+          similarReflectionGroups(reflectionId: $reflectionId, searchQuery: $searchQuery) {
+            id
+            ...ReflectionGroup_reflectionGroup
+          }
+          meeting(meetingId: $meetingId) {
+            ... on RetrospectiveMeeting {
+              ...DraggableReflectionCard_meeting
+              ...ReflectionGroup_meeting
+              id
+              teamId
+              localPhase {
+                phaseType
+              }
+              localStage {
+                isComplete
+                phaseType
+              }
+              phases {
+                phaseType
+                stages {
+                  isComplete
+                  phaseType
+                }
+              }
+              spotlightReflection {
+                id
+                ...DraggableReflectionCard_reflection
+              }
+            }
+          }
         }
-        meeting(meetingId: $meetingId) {
-          ...SpotlightGroups_meeting
-        }
-      }
-    `,
-    props.viewer
-  )
-  const meetingData = useFragment(
-    graphql`
-      fragment SpotlightGroups_meeting on RetrospectiveMeeting {
-        ...ReflectionGroup_meeting
       }
     `,
-    props.meeting
+    queryRef,
+    {UNSTABLE_renderPolicy: 'full'}
   )
-  const {similarReflectionGroups} = userData
+  const {viewer} = data
+  const {meeting, similarReflectionGroups} = viewer
+  const resultsRef = useRef<HTMLDivElement>(null)
+  const groupMatrix = useGroupMatrix(similarReflectionGroups, resultsRef, phaseRef)
+  const scrollHeight = useResultsHeight(resultsRef)
 
-  if (!similarReflectionGroups.length) {
-    return <SpotlightGroupsEmptyState />
-  }
+  if (!similarReflectionGroups.length) return <SpotlightGroupsEmptyState />
   return (
-    <Container>
-      {similarReflectionGroups.map((reflectionGroup) => {
-        return (
-          <ReflectionGroup
-            key={reflectionGroup.id}
-            meeting={meetingData}
-            phaseRef={phaseRef}
-            reflectionGroup={reflectionGroup}
-          />
-        )
-      })}
-    </Container>
+    <SimilarGroups>
+      <Scrollbar height={scrollHeight} ref={resultsRef}>
+        {groupMatrix.map((row) => (
+          <Column key={`row-${row[0].id}`}>
```

Comment:
`row[0].id` can be the same as the `group.id` key below. Differentiate with `row-` prefix